extensions/amp-selector/1.0/README.md

```diff
@@ -0,0 +1,287 @@
+```
+
+Alternatively, you may also make the light-weight pre-upgrade styles available inline:
+
+```html
+<style data-bento-boilerplate>
+ BentoSelector{
+    display: block;
+    overflow: hidden;
+    position: relative;
+  }
+</style>
+```
+
+#### Attributes on `<BentoSelector>`
+
+**disabled, form, multiple, name**
+
+The attributes above behave the same way as they do on a standard HTML [`<select>`](https://developer.mozilla.org/en/docs/Web/HTML/Element/select) element.
+
+**keyboard-select-mode**
+
+The `keyboard-select-mode` attribute dictates the keyboard navigation behavior for options inside `<BentoSelector>`.
+
+<ul>
+  <li>
+    `none` (default): The tab key changes focus between items in the `<BentoSelector>`. The user must press enter or space to change the selection. Arrow keys are disabled.
+  </li>
+  <li>
+    `focus`: Tab key gives focus to `<BentoSelector>`. The user navigates between items with the arrow keys. Must press space or enter to change the selection.
+  </li>
+  <li>
+    `select`: Tab key gives focus to `<BentoSelector>`. The selection changes as the user navigates options with arrow keys.
+  </li>
+</ul>
+
+#### Attributes on `<BentoSelectorOption>`
+
+**option**
+
+Indicates that the option is selectable. If a value is specified, the contents of the value is submitted with the form.
+
+**disabled, selected**
+
+The attributes above behave the same way as they do on a standard HTML [`<option>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/option) element.
+
+#### Actions
+
+The `BentoSelector` API allows you to perform the following actions:
+
+**selectBy(delta: number)**
+Closes the selector.
+
+```js
+api.selectBy(1); // Select next option in DOM sequence.
+api.selectBy(-2); // Select the option that is two previous in DOM sequence.
+```
+
+**toggle(optionValue: string, opt_select: boolean|undefined)**
+Toggles the option with the given `optionValue` to be selected or deselected based on `opt_select`. If `opt_select` is not present, then the option will be selected if currently not selected, and deselected if currently selected.
+
+```js
+api.toggle("a"); // Toggle the item with the attribute `option="a"`.
+api.toggle("1", true); // Select the item with the attribute `option="1"`.
+```
+
+#### Events
+
+The `BentoSelector` API allows you to register and respond to the following events:
+
+**select**
+
+This event is triggered when the user selects an option.
+Multi-selectors and single-selectors fire this when selecting or unselecting options.
+Tapping disabled options does not trigger the `select` event.
+
+<ul>
+  <li>
+  `event.data.targetOption` contains the `option` attribute value of the selected element.</li>
+  <li>
+  `event.data.selectedOptions` contains an array of the `option` attribute values of all selected elements.
+  </li>
+</ul>
+
+```js
+selector.addEventListener("select", (e) => console.log(e.data.targetOption))
+```
```

Comment:
the event listener in the preact world is the `onChange` prop, here's a similar example from base-carousel's readme:
```md
##### Events

The `BentoBaseCarousel` API allows you to register and respond to the following events:

**onSlideChange**

This event is triggered when the index displayed by the carousel has changed.

``jsx
<BentoBaseCarousel onSlideChange={(index) => console.log(index)}>
  <img src="puppies.jpg" />
  <img src="kittens.jpg" />
  <img src="hamsters.jpg" />
</BentoBaseCarousel>
``
```